frontend/src/v5/helpers/download.helper.ts

```diff
@@ -19,13 +19,15 @@ import axios from 'axios';
 import { isApiUrl } from '../services/api/default';
 
 const CachedURL: Record<string, string>  = {};
+let filenameFromRes: string = '';
 
 export const downloadAuthUrl = async (url):Promise<string> => {
 	if ( !isApiUrl(url) ) return url;
 
 	if (!CachedURL[url]) {
 		const response = await axios.get(url, { responseType: 'blob' });
 		CachedURL[url] = URL.createObjectURL(response.data);
+		filenameFromRes = response.headers['content-disposition']?.split('filename=')[1];
```

Comment:
It may be a good idea to store this in a second dictionary that goes along with `CachedURL` like `filenames`